### PR TITLE
Move IOperation API entries back to PublicAPI.Unshipped.txt

### LINKED ISSUE #18506: IOperation API entries should stay in PublicAPI.Unshipped.txt
https://github.com/dotnet/roslyn/pull/18434

Seems like this PR incorrectly moved unshipped IOperation APIs to PublicAPI.Shipped.txt. 

Anything that touches the IOperation API surface should be unshipped:
  1. OperationKind
  2. IOperation API surface in MS.CA.Semantics namespace (which includes the IOperation interfaces, OperationWalker, OperationVisitor, etc.) 
  3. SemanticModel.GetOperation API.
  4. RegisterOperationAction
  5. OperationAnalysisContext

### PR DESCRIPTION
**Pull request by mavasani**
Fixes #18506

### CODE REVIEW

**Comment by mavasani** [comment]
Tagging @dotnet/analyzer-ioperation @jaredpar @sharwell

**Comment by mavasani** [comment]
Tagging @dotnet/roslyn-compiler

File: src/Compilers/Core/Portable/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -3,11 +3,912 @@
 *REMOVED*virtual Microsoft.CodeAnalysis.Semantics.OperationVisitor<TArgument, TResult>.VisitCatch(Microsoft.CodeAnalysis.Semantics.ICatchClause operation, TArgument argument) -> TResult
```
Comment (genlu) [inline src/Compilers/Core/Portable/PublicAPI.Unshipped.txt]: I think this line can be removed now.

**Comment by genlu** [comment]
👍 Besides the comment on *removed* entry

File: src/Compilers/Core/Portable/PublicAPI.Unshipped.txt (hunk 1 above)
Comment (mavasani) [inline src/Compilers/Core/Portable/PublicAPI.Unshipped.txt]: Thanks, done.

**Comment by jcouv** [comment]
@mavasani Can you clarify the context? Were those APIs accidentally moved to the "shipped" file?

FYI @jaredpar @gafter since they requested a ping on any public API changes.

**Comment by mavasani** [comment]
@jcouv - Yes, they were accidentally moved when we bulk moved unshipped APIs to shipped files after 2.0.0 release.

**Pull request merged**